web/scripts/widgets.js

```diff
@@ -10,37 +10,121 @@ function getNumberDefaults(inputData, defaultStep) {
 	return { val: defaultVal, config: { min, max, step: 10.0 * step } };
 }
 
-export function addRandomizeWidget(node, targetWidget, name, defaultValue = false) {
-	const randomize = node.addWidget("toggle", name, defaultValue, function (v) {}, {
-		on: "enabled",
-		off: "disabled",
+export function addValueControlWidget(node, targetWidget, defaultValue = "randomize", values) {
+	const valueControl = node.addWidget("combo", "control_after_generate", defaultValue, function (v) { }, {
+		values: ["fixed", "increment", "decrement", "randomize"],
 		serialize: false, // Don't include this in prompt.
-	});
+	})
+	valueControl.afterQueued = () => {
+
+		var v = valueControl.value;
+		var w = targetWidget.name;
+
+		let min = 0.0;
+		let max = targetWidget.options?.max;
+		let range = Math.max(min, max);
+
+		//adjust values based on valueControl Behaviour
+		switch (v) {
+			case ("fixed"):
+				console.log("fixed");
+				break;
+			case ("increment"):
+				if (min != null || max != null) {
+					/*if (max) { //loop to lowest and continue batch
+						targetWidget.value = 0;
+						console.log("increment");
+					} else {
+						targetWidget.value += 1;
+						console.log("increment");
+					}
+				} else {*/
+					if (max) {
+						if (w == "denoise") {
+							targetWidget.value += 0.1;
+							console.log("denoise decrement");
+						} else if (w == "cfg") {
+							targetWidget.value += 0.5;
```

Comment:
Why can't you increment or decrement using the step that is already defined in the widget?